File: Ridy-taxi-app/android/common/src/main/java/com/innomalist/taxi/common/networking/http/GetStripeClientSecret.kt
package com.innomalist.taxi.common.networking.http

import com.innomalist.taxi.common.models.Rider
import com.innomalist.taxi.common.networking.http.interfaces.HTTPRequest

class GetStripeClientSecret(gatewayId: Int, amount: Int, currency: String): HTTPRequest() {
    override val path: String = "stripe_client_secret"
    init {
        this.params = mapOf("gatewayId" to gatewayId.toString(), "amount" to amount, "currency" to currency)
    }
}

data class StripeClientSecretEndpointResult(
        val clientSecret: String
)

File: Ridy-taxi-app/ios/common/Networking/HTTP/GetStripeClientSecret.swift
//
//  GetStripeClientSecret.swift
//  driver
//
//  Created by Manly Man on 6/2/20.
//  Copyright © 2020 minimal. All rights reserved.
//

import Foundation

public class GetStripeClientSecret: HTTPRequest {
    public var path: String = "stripe_client_secret"
    
    public typealias ResponseType = StripeClientSecretResult
    public var params: [String : Any]?
    
    public init(gatewayId: Int, amount: Int, currency: String) {
        self.params = [
            "gatewayId": gatewayId,
            "amount": amount,
            "currency": currency
        ]
    }
}

public struct StripeClientSecretResult: Codable {
    var clientSecret: String
}
